VxWorks/Os/Queue.cpp

```diff
@@ -31,6 +31,16 @@ QueueInterface::Status VxWorksQueue::send(const U8* buffer,
         return QueueInterface::Status::UNINITIALIZED;
     }
 
+    // VxWorks queues do not support true priority.  Priority of 1 or 0 is only allowed in cases where an ISR needs to
+    // be shuttled to the front of the line. This is because prioritized messages in VxWorks are in LIFO order and only
+    // support a boolean priority, which violates key expectations of Os::Queue. This is allowed in interrupt context as
+    // interrupts typically need to be processed with high priority.
+    if (!((intContext() && priority < 2) || priority == 0)) {
+        return QueueInterface::Status::NOT_SUPPORTED;
+    }
+
+    // Should not block in interrupt context.
```

Comment:
Make this stronger: "It is illegal to block while in interrupt context"

Reply:
Should we allow a blocking call if it would not have blocked?

Reply:
No. Let's catch this error as soon as possible